### PR TITLE
Window filter

### PR DESCRIPTION
**Pull request by Joe-Joe-Joe-Joe**
- replaced discrete lowpass filter with a moving average filter. Seems to be more robust
- Tested processor and charging in this configuration

### CODE REVIEW

File: STM32Cube/Tasks/trajectory.c [hunk 1]
```diff
@@ -63,7 +64,7 @@ void trajectory_task(void* argument) {
     float prev_time = -1;
     int32_t prev_alt = INT_MAX;
     uint32_t recovery_timeout = 0;
-    float filter_data[2] = {0.0f}; //storage for lowpass filter data
+    float filter_data[3] = {0.0f}; //storage for lowpass filter data
```
Comment (patrick-gu) [inline STM32Cube/Tasks/trajectory.c:67]: this is slightly misleading

**Review by celery6 — APPROVED** [review]

File: STM32Cube/Tasks/trajectory.c [hunk 2]
```diff
@@ -73,7 +74,10 @@ void trajectory_task(void* argument) {
                 if (prev_alt != INT_MAX) {
                     float vely_raw = (altTime.alt - prev_alt) * 1000.0 /
                                  (altTime.time - prev_time);
-                    float vely = second_order_lowpass_filter(vely_raw, filter_data, VELOCITY_FILTER_ALPHA);
+
+                    //float vely = second_order_lowpass_filter(vely_raw, filter_data, VELOCITY_FILTER_ALPHA);
+                    float vely = moving_average_filter(vely_raw, filter_data);
```
Comment (patrick-gu) [inline STM32Cube/Tasks/trajectory.c:79]: I think this might effectively increase the number of timeout readings, which I assume is intended

**Review by patrick-gu — APPROVED** [review]
LGTM

File: STM32Cube/Tasks/trajectory.c (hunk 2 above)
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Tasks/trajectory.c:79]: yes
should reduce false positives during coast

**Pull request merged**